### DISCUSSION TITLE
Explanation on new parameters of listener in ref.listen?

### DISCUSSION THREAD

**Discussion by Ruzo** [rrousselGit/riverpod#926] [Q&A] [answered]
Hi,

Is there an explanation somewhere about the nullable and non-nullable parameters in the new version of `listener` in `ref.listen`?

```dart
void listen<T>(
  ProviderListenable<T> provider,
  void Function(T?, T) listener, {
  void Function(Object, StackTrace)? onError,
})
```

Thanks!

**Comment by daniellampl** [comment, marked as answer]
The first parameter is the previous value and the second is the current value of the listened provider.

**Reply by Ruzo** [reply]
Ah, okay. Thanks a lot!

**Comment by ThomasKaputa** [comment]
Shouldn't the Documentation get updated?
There are examples with ref.listen with just one parameter which don't compile anymore. e.q.
final counterProvider = StateNotifierProvider<Counter, int>((ref) => Counter());

final anotherProvider = Provider((ref) {
  ref.listen<int>(counterProvider, (int count) {
    print('The counter changed ${count}');
  });
  ...
});

in https://riverpod.dev/docs/concepts/reading/
